src/main/java/bowlingexercise3/Bowling.java

```diff
@@ -46,7 +46,10 @@ public int score() {
         // distinguishing it between a spare
         for (int frame = 0; frame < 10; frame++) {
 
-            if (spare(cursor)) {
+            if (Strike(cursor)) {
+                score += 10 + turns[cursor+1] + turns[cursor+2];
+                cursor ++;
+            } else if (spare(cursor)) {
                 score += 10 + turns[cursor + 2];
```

Comment:
If you score a spare, should it be the addition of the current score, plus the next one, so score += 10 + turns[cursor + 1]; rather than 2? I wasn't sure about some of the scores in the tests, which lead me to look further, but I'm not expert on bowling!!

Reply:
The cursor is essentially a pointer and it moves forward two points on the string to discard the pipe. If I do what you suggest, the overall score doesn't add up correctly.

Reply:
> The cursor is essentially a pointer and it moves forward two points onto the string to discard the pipe. If I do what you suggest, the overall score doesn't add up correctly.

ok - I'm getting a bit lost - I think the population of the array starts at index 1 rather than 0, so it needs to add 1. In simplistic terms I' thinking a strike adds the next 2 balls and the spare only the next ball, so I don't quite follow it!